src/devices/Tca955x/Tca955x.cs

```diff
@@ -0,0 +1,519 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Device.Gpio;
+using System.Device.I2c;
+using System.Runtime.CompilerServices;
+using System.Threading;
+
+namespace Iot.Device.Tca955x
+{
+    /// <summary>
+    /// Base class for the Tca55x I2C I/O Expander
+    /// </summary>
+    public abstract class Tca955x : GpioDriver
+    {
+        private readonly int _interrupt;
+        private readonly Dictionary<int, PinValue> _pinValues = new Dictionary<int, PinValue>();
+        private readonly ConcurrentDictionary<int, PinChangeEventHandler> _eventHandlers = new ConcurrentDictionary<int, PinChangeEventHandler>();
+        private readonly Dictionary<int, PinEventTypes> _interruptPins = new Dictionary<int, PinEventTypes>();
+        private readonly Dictionary<int, PinValue> _interruptLastInputValues = new Dictionary<int, PinValue>();
+
+        private GpioController? _controller;
+
+        private bool _shouldDispose;
+
+        private byte _adress = 0x20;
+        private I2cDevice _bus;
+
+        private object _interruptHandlerLock = new object();
+
+        private ushort _gpioOutputCache;
+
+        /// <summary>
+        /// Constructor for the Tca9555 I2C I/O Expander.
+        /// </summary>
+        /// <param name="bus">The I2C Bus the device is connected to.</param>
+        /// <param name="interrupt">The input pin number that is connected to the interrupt.</param>
+        /// <param name="gpioController">The controller for the reset and interrupt pins. If not specified, the default controller will be used.</param>
+        /// <param name="shouldDispose">True to dispose the Gpio Controller.</param>
+        /// <param name="adress">
+        /// The device address for the connection on the I2C bus.
+        /// Start with 0x20 and ends with 0x27
+        /// </param>
+        protected Tca955x(I2cDevice bus, int interrupt = -1, GpioController? gpioController = null, bool shouldDispose = true, byte adress = 0x20)
+        {
+            _bus = bus;
+            _interrupt = interrupt;
+            _adress = adress;
+
+            if (adress < 0x20 ||
```

Comment:
you can then use the const and a +7 if that's the range. A few less magic numbers then.